plan/eliminate_projection.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/model"
+)
+
+// EliminateProjection eliminate projection operator to avoid the cost of memory copy in the iterator of projection.
+func EliminateProjection(p LogicalPlan, reorder bool, orderedSchema expression.Schema) LogicalPlan {
```

Comment:
If you encounter a query like "select b.id, a.id, b.c, a.c from a,b where a.id = b.id", Can you do this optimization?